bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/BusyIndicator.java

```diff
@@ -70,6 +70,83 @@ public static void showWhile(Display display, Runnable runnable) {
 		}
 	}
 
+	/**
+	 * If called from a {@link Display} thread, waits for the given
+	 * <code>Future</code> to complete and provides busy feedback using the
+	 * busy indicator. While waiting for completion, pending UI events are processed
+	 * to prevent UI freeze.
+	 *
+	 * If there is no {@link Display} for the current thread, the {@link Future#get()} will be executed
+	 * ignoring any {@link ExecutionException} and no busy feedback will be displayed.
+	 *
+	 * In any case, the passed {@link Future} is returned for chaining of further operations.
+	 *
+	 * @param future the {@link Future} for which busy feedback is to be shown.
+	 * @since 3.127
+	 */
+	public static void showWhile(Future<?> future) {
+		if (future == null) {
+			SWT.error(SWT.ERROR_NULL_ARGUMENT);
+		}
+		if (!future.isDone()) {
+			Display display = Display.getCurrent();
+			if (display == null) {
```

Comment:
```suggestion
			if (display == null || display.isDisposed()) {
```

Reply:
in case eithe display is disposed, should this maybe better lead to a failure?
e.g. if I look at the old implementation, there the display is not tested for disposal and the caller must ensure this.